File: client/app/src/main/java/me/shouheng/notepal/viewmodel/NotebookViewModel.java
package me.shouheng.notepal.viewmodel;

import android.arch.lifecycle.LiveData;

import me.shouheng.notepal.model.Notebook;
import me.shouheng.notepal.model.data.Resource;
import me.shouheng.notepal.model.enums.ItemStatus;
import me.shouheng.notepal.repository.BaseRepository;
import me.shouheng.notepal.repository.NotebookRepository;

/**
 * Created by shouh on 2018/3/17.
 */
public class NotebookViewModel extends BaseViewModel<Notebook> {

    @Override
    protected BaseRepository<Notebook> getRepository() {
        return new NotebookRepository();
    }

    public LiveData<Resource<Notebook>> update(Notebook notebook, ItemStatus fromStatus, ItemStatus toStatus) {
        return ((NotebookRepository) getRepository()).update(notebook, fromStatus, toStatus);
    }
}

File: client/app/src/main/java/me/urakalee/next2/viewmodel/NotebookViewModel.kt
package me.urakalee.next2.viewmodel

import android.arch.lifecycle.LiveData
import me.shouheng.notepal.model.Notebook
import me.shouheng.notepal.model.data.Resource
import me.shouheng.notepal.repository.NotebookRepository
import me.shouheng.notepal.viewmodel.BaseViewModel

/**
 * @author Uraka.Lee
 */
class NotebookViewModel : BaseViewModel<Notebook>() {

    override fun getRepository(): NotebookRepository {
        return NotebookRepository()
    }

    fun list(): LiveData<Resource<List<Notebook>>> {
        return repository.get(null, null)
    }

    fun delete(notebook: Notebook): LiveData<Resource<Notebook>> {
        return repository.delete(notebook)
    }
}
